wagtail/contrib/wagtailfrontendcache/backends.py

```diff
@@ -84,3 +88,49 @@ def purge(self, url):
         if response_json['result'] == 'error':
             logger.error("Couldn't purge '%s' from Cloudflare. Cloudflare error '%s'", url, response_json['msg'])
             return
+
+
+class CloudfrontBackend(BaseBackend):
+    def __init__(self, params):
+        self.client = boto3.client('cloudfront')
+        try:
+            self.cloudfront_distribution_id = params.pop('DISTRIBUTION_ID')
+        except KeyError:
+            raise ImproperlyConfigured(
+                "The setting 'WAGTAILFRONTENDCACHE' requires the object "
+                "'DISTRIBUTION_ID'."
+            )
+
+    def purge(self, url):
+        url_parsed = urlparse(url)
+        distribution_id = None
+
+        if isinstance(self.cloudfront_distribution_id, dict):
+            host = url_parsed.hostname
+            if host in self.cloudfront_distribution_id:
+                distribution_id = self.cloudfront_distribution_id.get(host)
+            else:
+                logger.error("Couldn't purge '%s', hostname '%s' not found in mapping", url, host)
```

Comment:
Good point, I'll keep it consistent with the other error messages (like CloudFlare)